agent/websockets/connection.go

```diff
@@ -252,3 +264,47 @@ func (conn *Connection) ReadServerMessages() ([]interface{}, error) {
 		return nil, nil
 	}
 }
+
+// injectWebsocketMessage injects a shim header value into a single websocket message in-place.
+// Returns a pointer to a new copy of the struct on success.
+func injectWebsocketMessage(msg *message, injectionPath []string, injectionValues map[string]string) (*message, error) {
+	if msg == nil || injectionValues == nil {
+		return nil, errors.New("unexpected nil message")
+	}
+	if injectionValues == nil {
+		return nil, errors.New("unexpected nil values map")
+	}
+	if len(injectionValues) == 0 {
```

Comment:
Do we really want to differentiate between a nil map and empty map? I thought in go they are usually handled same

Reply:
I think you have a point, it might be easier to reason about if passing a nil map in also simply sends the message, though technically a nil map should never be an expected input.